src/cobra/core/gene.py

```diff
@@ -559,8 +655,99 @@ def _repr_html__(self):
             gpr=format_long_string(self.to_string(), 100)
         )
 
-    # def as_symbolic(self):
-    #     # ...
+    def as_symbolic(
+        self,
+        names: dict = None,
+    ) -> Union[sp_Or, sp_And, Symbol]:
+        """Convert compiled ast to sympy expression.
+
+        Parameters
+        ----------
+        self : GPR
+           compiled ast Module describing GPR
+        names: dict
+           dictionary of gene ids to gene names. If this is empty,
+           returns sympy expression using gene ids
+
+        Returns
+        ------
+        Symbol or BooleanFunction
+            SYMPY expression (Symbol or And or Or). Symbol("") if the GPR is empty
+
+        Notes
+        -----
+        Calls _symbolic_gpr()
+        """
+        # noinspection PyTypeChecker
+        if names:
+            GPRGene_dict = {gid: Symbol(names[gid]) for gid in self.genes}
+        else:
+            GPRGene_dict = None
+        return self._symbolic_gpr(self, GPRGene_dict=GPRGene_dict)
+
+    def _symbolic_gpr(
+        self,
+        expr: Union["GPR", Expression, BoolOp, Name, list] = None,
+        GPRGene_dict: dict = None,
+    ) -> Union[sp_Or, sp_And, Symbol]:
+        """Parse gpr into SYMPY using ast similar to _ast2str()
+
+        Parameters
+        ----------
+        expr : AST or GPR or list or Name or BoolOp
+            compiled GPR
+        GPRGene_dict: dict
+            dictionary from gene id to GPRGeneSymbol
+        Returns
+        -------
+        Symbol or BooleanFunction
+            SYMPY expression (Symbol or And or Or). Symbol("") if the GPR is empty
+        """
+        if GPRGene_dict is None:
+            GPRGene_dict = {gid: Symbol(name=gid) for gid in expr.genes}
+        if isinstance(expr, (Expression, GPR)):
+            return (
+                self._symbolic_gpr(expr.body, GPRGene_dict) if expr.body else Symbol("")
+            )
+        else:
+            if isinstance(expr, Name):
+                return GPRGene_dict.get(expr.id)
+            elif isinstance(expr, BoolOp):
+                op = expr.op
+                if isinstance(op, Or):
+                    # noinspection PyTypeChecker
+                    sym_exp = sp_Or(
+                        *[self._symbolic_gpr(i, GPRGene_dict) for i in expr.values]
+                    )
+                elif isinstance(op, And):
+                    # noinspection PyTypeChecker
+                    sym_exp = sp_And(
+                        *[self._symbolic_gpr(i, GPRGene_dict) for i in expr.values]
+                    )
+                else:
+                    raise TypeError("unsupported operation " + op.__class__.__name)
+                return sym_exp
+            elif expr is None or (isinstance(expr, list) and len(expr) == 0):
```

Comment:
might even do `elif not expr:`. This will behave the same for the two cases and catch some others.